src/Accordion/Accordion.tsx

```diff
@@ -1,95 +1,61 @@
 import React from "react";
 import { randomId } from "@sebgroup/frontend-tools/dist/randomId";
-import { isPrimitive } from "@sebgroup/frontend-tools/dist/isPrimitive";
 import classnames from "classnames";
-import { AccordionItemBody } from "./AccordionItemBody";
+import { AccordionItemProps, AccordionItem } from "./AccordionItem";
 import "./accordion.scss";
 
-export interface AccrodionListItem {
-    /** The header of the accordion item */
-    header: React.ReactNode;
-    /** A sub-header description rendered under the header */
-    subHeader?: React.ReactNode;
-    /** The content of the accordion item */
-    content: React.ReactNode;
-}
-
-export interface AccordionProps {
-    /** Element class name */
-    className?: string;
-    /** Element id */
-    id?: string;
+export type AccordionProps = React.DetailedHTMLProps<React.HTMLAttributes<HTMLDivElement>, HTMLDivElement> & {
     /**
      * The accordion list of items
-     * @see AccordionListItem interface
+     * @see AccordionItemProps interface
      */
-    list: Array<AccrodionListItem>;
+    list?: Array<AccordionItemProps>;
     /** An alternative version of the accordion */
     alternative?: boolean;
-    /** The index of an item that would be set to be expanded by default  */
-    defaultExpanded?: number;
-}
-
-const Accordion: React.FC<AccordionProps> = React.memo((props: AccordionProps) => {
-    const [active, setActive] = React.useState<number>(props.defaultExpanded);
-    const [idList, setIdList] = React.useState<Array<string>>([]);
-    const [id, setId] = React.useState<string>("");
-    const [className, setClassName] = React.useState<string>("");
-
-    React.useEffect(() => constructIds(), [props.list]);
-    React.useEffect(() => setActive(props.defaultExpanded), [props.defaultExpanded]);
+    /** An event handler triggered when an accordion toggle is clicked */
+    onToggle?: React.MouseEventHandler<HTMLButtonElement>;
+};
+
+export const Accordion: React.FC<AccordionProps> = React.memo(({ list, alternative, onToggle, ...props }: AccordionProps) => {
+    const [active, setActive] = React.useState<number>(Number(props.defaultValue));
+    const [id, setId] = React.useState<string>(props.id);
+    const [className, setClassName] = React.useState<string>("seb accordion");
+
+    React.useEffect(() => {
+        let defaultChecked: number;
+        defaultChecked = list?.findIndex((item: AccordionItemProps) => item.defaultChecked);
+        defaultChecked = React.Children.toArray(props.children).findIndex((Child: React.ReactElement<AccordionItemProps>) => Child.props?.defaultChecked);
```

Comment:
you changed the value of defaultChecked twice(Line26 and Line 27) without using it elsewhere, i believe the first line is not required anymore?
can you double check.

Reply:
🤔

Reply:
It is used